crates/burn-core/src/module/base.rs

```diff
@@ -97,17 +97,18 @@ pub trait Module<B: Backend>: Clone + Send + core::fmt::Debug {
     ///
     /// # Notes
     ///
-    /// This is similar to [to_device](Module::to_device), but it ensures the module will
-    /// have its own autodiff graph.
+    /// This is similar to [to_device](Module::to_device), but it ensures the output module on the
+    /// new device will have its own autodiff graph.
     fn fork(self, device: &B::Device) -> Self;
 
     /// Move the module and all of its sub-modules to the given device.
     ///
     /// # Warnings
     ///
-    /// The device operations will be registered in the autodiff graph. Therefore, be sure to call
-    /// backward only one time even if you have the same module on multiple devices. If you want to
-    /// call backward multiple times, look into using [fork](Module::fork) instead.
+    /// The operation supports autodiff and it will be registered when activated. However, this may
+    /// not be what you want. The output model will be an intermediary model, meaning that you
```

Comment:
What's an "intermediary model"? We might want to clarify 🤔

Reply:
I took the term to mirror `intermediary tensor` when computing backprop. Which is a node that has gradients, but only as a mean to compute other gradients, and therefore their value is flushed during the backward calculation. Not sure how can I make it more clear without defining how autodiff works 😅

Reply:
Ahhh ok! I get it now.

Yeah we can keep it like that.